src/Components/Nav.jsx

```diff
@@ -0,0 +1,23 @@
+import { useEffect } from "react";
+import { useState } from "react";
+import { getCategories } from "../Utils/api";
+import { Link } from "react-router-dom";
+
+export const Nav = ({ setCategories, categories }) => {
+  useEffect(() => {
+    getCategories().then(({ categories }) => {
+      setCategories(categories);
+    });
+  }, [categories]);
```

Comment:
this dependency array will cause an infinite loop.  Do we need a dependency here?  Or do we want categories to be set once on mounting?